json_parser.py

```diff
@@ -15,21 +15,21 @@ def get_path(self) -> str:
 
     def _read_json_file(self):
         """Reads the content of a JSON file located in _path and deserializes its content."""
-        with open(self._path) as file:
+        with open(self._path) as file: #what if path doesn't exist?
             file_content = file.read()
             return json.loads(file_content)
 
-    def scan_translation_keys(self):
+    def scan_translation_keys(self): #in general, the more nested structures, the more complex the code
```

Comment:
I rewrote the method, it uses now a supplementary recursive one. I pass the maximum nested level as a required parameter.